File: 05-may/11-separate-the-digits-in-an-array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var separateDigits = function(nums) {
    const answer = [];
    for (const num of nums) {
        for (const ch of String(num)) {
            answer.push(ch.charCodeAt(0) - 48);
        }
    }
    return answer;
};

File: 05-may/11-separate-the-digits-in-an-array/solution.java
class Solution {
    public int[] separateDigits(int[] nums) {
        int size = 0;
        for (int num : nums) {
            size += String.valueOf(num).length();
        }

        int[] answer = new int[size];
        int index = 0;
        for (int num : nums) {
            String digits = String.valueOf(num);
            for (int i = 0; i < digits.length(); i++) {
                answer[index++] = digits.charAt(i) - '0';
            }
        }
        return answer;
    }
}
